runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -340,16 +339,14 @@ private void addReconnectionOnCloseFuture(@Nonnull Channel channel,
      * @param bootstrap         The channel boostrap to use
      * @return                  A {@link ChannelFuture} which is c
      */
-    private ChannelFuture connectAsync(@Nonnull Bootstrap bootstrap) {
+    private void connectAsync(@Nonnull Bootstrap bootstrap) {
         // If shutdown, return a ChannelFuture that is exceptionally completed.
         if (shutdown) {
-            return new DefaultChannelPromise(channel, GlobalEventExecutor.INSTANCE)
```

Comment:
Does this only return an exceptionally completed channel promise? It doesn't close the `channel` right?

Reply:
No It does't close the channel.